.github/workflows/lint.yml

```diff
@@ -0,0 +1,64 @@
+name: Lint
+
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**.md'
+  pull_request:
+    branches:
+      - main
+    paths-ignore:
+      - '**.md'
+    types:
+      - opened
+      - reopened
+      - synchronize
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  group: ${{ github.ref_name }}-lint
+  cancel-in-progress: true
+
+jobs:
+
+  lint:
+    name: Lint
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+      - name: Setup Golang Environment
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: go.mod
+      - name: Lint Code
+        uses: golangci/golangci-lint-action@v3.2.0
+        with:
+          args: --timeout 10m0s
+          only-new-issues: true
```

Comment:
I've added `only-new-issues` that makes the action fail only if there are lint errors in the changes made by the PR, instead of running on the entire codebase, let me know if that works

Reply:
seems good to me